pkg/operator/controllers/muo/deploy_test.go

```diff
@@ -0,0 +1,126 @@
+package muo
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/golang/mock/gomock"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	mock_dynamichelper "github.com/Azure/ARO-RP/pkg/util/mocks/dynamichelper"
+)
+
+func TestDeployCreateOrUpdate(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	setPullSpec := "MyMUOPullSpec"
+	cluster := &arov1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: arov1alpha1.SingletonClusterName,
+		},
+		Spec: arov1alpha1.ClusterSpec{
+			OperatorFlags: arov1alpha1.OperatorFlags{
+				pullSpec: setPullSpec,
+			},
+		},
+	}
+
+	k8scli := fake.NewSimpleClientset()
+	dh := mock_dynamichelper.NewMockInterface(controller)
+
+	// When the DynamicHelper is called, store the object kinds it creates.
+	var deployments []*appsv1.Deployment
+	deployedObjects := make(map[string]bool)
+	accessor := meta.NewAccessor()
+	check := func(ctx context.Context, objs ...kruntime.Object) error {
+		for _, i := range objs {
+			kind, err := accessor.Kind(i)
+			if err != nil {
+				t.Error(err)
+			}
+			if d, ok := i.(*appsv1.Deployment); ok {
+				deployments = append(deployments, d)
+			}
+			deployedObjects[kind] = true
```

Comment:
We deploy multiple roles. We might accidentaly deelete one of the role yamls and not catch it.

Maybe let's use a slice of kinds? This way we make sure that we deploy the right number of objects of right kinds, but don't go in to too much details. Trying to find a balance here. What do you think?

Reply:
I went with a map of kind: count, which seems simple and clear. Considering making sure we put things in the right namespaces, but I think that's maybe out of scope of this (since we may well need to put things in various namespaces if MUO needs it)